### DISCUSSION TITLE
Only the first frame is decoded successfully!

### DISCUSSION THREAD

**Discussion by xlievo** [ralfbiedert/openh264-rs#56] [General]
Only the first frame decoded successfully, subsequent frames failed, probably what is the reason? 
Does it have something to do with initialization Settings or capacity?
It is worth mentioning that all frames can be successfully decoded with other openh264 wrappers.

In addition, can you set some parameters for initialization?
I hope to align with other openh264 wrapper libraries to further find out why.

**Comment by xlievo** [comment]
Close the issue temporarily until someone brings it up again !

**Comment by ralfbiedert** [comment]
It's hard to tell without the actual .h264 file. I recommend to download a H264 Analyzer and look at your stream. Posting the code you used would be good, too.

**Comment by xlievo** [comment]
With decoder_ref.raw_api.decode_frame2, there is no problem with decoding.
Is it possible to add decode_frame2 decoder function, and the following construction parameters
```rust
impl DecoderConfig

pub fn e_videobstype(mut self, e_videobstype: openh264_sys2::VIDEO_BITSTREAM_TYPE) -> Self {
    self.params.sVideoProperty.eVideoBsType = e_videobstype;
    self
}

pub fn e_ecactiveidc(mut self, e_ecactiveidc: ERROR_CON_IDC) -> Self {
    self.params.eEcActiveIdc = e_ecactiveidc;
    self
}
```

**Comment by ralfbiedert** [comment]
Generally yes. This might require some verification wrt what these parameters do, but feel free to file a PR.
